client/client.go

```diff
@@ -233,3 +234,19 @@ func ParseHost(host string) (string, string, string, error) {
 	}
 	return proto, addr, basePath, nil
 }
+
+// CustomHTTPHeaders returns the custom http headers associated with this
+// instance of the Client. This operation doesn't acquire a mutex.
+func (cli *Client) CustomHTTPHeaders() map[string]string {
+	m := make(map[string]string)
+	for k, v := range cli.customHTTPHeaders {
+		m[k] = v
+	}
+	return m
+}
+
+// UpdateCustomHTTPHeaders updates the custom http headers associated with this
+// instance of the Client. This operation doesn't acquire a mutex.
+func (cli *Client) UpdateCustomHTTPHeaders(headers map[string]string) {
```

Comment:
Could we call this `SetCustomHTTPHeaders` instead? Being named `Update...` I was assuming it was going to merge both maps at first